pkg/api/register.go

```diff
@@ -111,6 +111,15 @@ func AddToScheme(scheme *runtime.Scheme) {
 		&unversioned.APIGroup{},
 		&unversioned.APIResourceList{},
 	)
+
+	// Types registered here will be kept unversioned when they are encoded.
+	// It's for backward compatibility with release 1.1. clients.
+	Scheme.AddToKeepUnverioned(
```

Comment:
For some of the new groups there isn't likely to be a v2 group at the
same time Kube core API goes to v2.  And community extensions will
likely not upgrade at the same time.  I don't know that there *will*
be a real v2 for all groups at the same time.  So we could wait, but
that might be a long time.